### ISSUE TITLE
Type definition / import doesn't marked as used because of lack of symbol occurrence

### ISSUE THREAD

**Issue by tanishiking** [tanishiking/scalafix-unused#5]
**Describe the bug**

- See: PrefixScala3.scala 
- https://github.com/lampepfl/dotty/blob/e7a641c5ad61fa683423954fa9263079a890c809/tests/semanticdb/expect/Prefixes.expect.scala#L27
  - This is caused by the lack of symbol in TypeTree

**Installation:**

- Scala version: v3.0.1

**Minimal Reproduciable code**

```scala
package fix.example
package prefixes3

class C {
  type T
  def m1: T = ???

  object N {
    type U
  }
  def k1: N.U = ???
}



object Test {
  val c: C = ???
  def m2: c.T = ???
  def k2: c.N.U = ???
  // TODO: fix c.N._ shouldn't be reported
  // this is because `U` doesn't have symbol
  // see: https://github.com/lampepfl/dotty/blob/e7a641c5ad61fa683423954fa9263079a890c809/tests/semanticdb/expect/Prefixes.expect.scala#L27
  // after fix the issue, merge Prefixes.scala and this file
  //
  import c.N._
  def k3: U = ???

  def n2: M.T = ???

  import M._
  def n3: T = ???
}

```

**Comment by tanishiking** [comment]
Related to https://github.com/lampepfl/dotty/issues/13270

**Comment by tanishiking** [comment]
fixed v3.0.2
